src/processor/py/mod.rs

```diff
@@ -0,0 +1,1091 @@
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        Ok(v.end_time_unix_nano)
+    }
+    #[setter]
+    fn set_end_time_unix_nano(&self, new_value: u64) -> PyResult<()> {
+        let mut v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        v.end_time_unix_nano = new_value;
+        Ok(())
+    }
+    // TODO
+    // pub attributes: Arc<Mutex<Vec<Arc<Mutex<KeyValue>>>>>,
+    #[getter]
+    fn dropped_attributes_count(&self) -> PyResult<u32> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        Ok(v.dropped_attributes_count)
+    }
+    #[setter]
+    fn set_dropped_attributes_count(&self, new_value: u32) -> PyResult<()> {
+        let mut v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        v.dropped_attributes_count = new_value;
+        Ok(())
+    }
+    // TODO
+    // //pub events: ::prost::alloc::vec::Vec<span::Event>,
+    #[getter]
+    fn dropped_events_count(&self) -> PyResult<u32> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        Ok(v.dropped_events_count)
+    }
+    #[setter]
+    fn set_dropped_events_count(&self, new_value: u32) -> PyResult<()> {
+        let mut v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        v.dropped_events_count = new_value;
+        Ok(())
+    }
+    // TODO
+    // //pub links: ::prost::alloc::vec::Vec<span::Link>,
+    #[getter]
+    fn dropped_links_count(&self) -> PyResult<u32> {
+        let v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        Ok(v.dropped_links_count)
+    }
+    #[setter]
+    fn set_dropped_links_count(&self, new_value: u32) -> PyResult<()> {
+        let mut v = self.inner.lock().map_err(|_| {
+            PyErr::new::<pyo3::exceptions::PyRuntimeError, _>("Failed to lock mutex")
+        })?;
+        v.dropped_links_count = new_value;
+        Ok(())
+    }
+    // pub status: Arc<Mutex<Option<Status>>>,
+}
+
+#[pyclass]
+struct LoggingStdout;
+
+#[pymethods]
+impl LoggingStdout {
+    fn write(&self, data: &str) {
+        println!("stdout from python: {:?}", data);
+    }
+}
+
+// Module initialization
+#[pymodule]
+pub fn rotel_python_processor_sdk(m: &Bound<'_, PyModule>) -> PyResult<()> {
+    m.add_class::<PyAnyValue>()?;
+    m.add_class::<PyKeyValue>()?;
+    m.add_class::<PyResource>()?;
+    m.add_class::<PyAttributes>()?;
+    m.add_class::<PySpan>()?;
```

Comment:
Do the `*Iter` and ScopeSpan types need to be included here?

Reply:
Not sure we'll need the iter, but yes eventually all the classes we want to allow users to import will need to be added.